Library/Homebrew/extend/os/mac/keg_relocate.rb

```diff
@@ -0,0 +1,143 @@
+class Keg
+  def fix_install_linkage
+    mach_o_files.each do |file|
+      file.ensure_writable do
+        change_dylib_id(dylib_id_for(file), file) if file.dylib?
+
+        each_install_name_for(file) do |bad_name|
+          # Don't fix absolute paths unless they are rooted in the build directory
+          next if bad_name.start_with?("/") && !bad_name.start_with?(HOMEBREW_TEMP.to_s)
+
+          new_name = fixed_name(file, bad_name)
+          change_install_name(bad_name, new_name, file) unless new_name == bad_name
+        end
+      end
+    end
+
+    symlink_files.each do |file|
+      link = file.readlink
+      # Don't fix relative symlinks
+      next unless link.absolute?
+      if link.to_s.start_with?(HOMEBREW_CELLAR.to_s) || link.to_s.start_with?(HOMEBREW_PREFIX.to_s)
+        FileUtils.ln_sf(link.relative_path_from(file.parent), file)
+      end
+    end
```

Comment:
This `symlink_files.each do` block is definitely not OS-specific and needs to be refactored into a separate method (and moved back into the generic part of the implementation).